jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -423,34 +431,49 @@ ecma_builtin_typedarray_prototype_map (ecma_value_t this_arg, /**< this argument
     return ecma_raise_type_error (ECMA_ERR_MSG ("Callback function is not callable."));
   }
 
-  ecma_object_t *obj_p = ecma_get_object_from_value (this_arg);
-  uint32_t len = ecma_typedarray_get_length (obj_p);
+  ecma_object_t *src_obj_p = ecma_get_object_from_value (this_arg);
+  ecma_typedarray_info_t src_info = ecma_typedarray_get_info (src_obj_p);
+
   ecma_object_t *func_object_p = ecma_get_object_from_value (cb_func_val);
   ecma_value_t ret_value = ECMA_VALUE_EMPTY;
 
-  ecma_value_t new_typedarray = ecma_op_create_typedarray_with_type_and_length (obj_p, len);
+  ecma_value_t new_typedarray = ecma_op_create_typedarray_with_type_and_length (src_obj_p, src_info.typedarray_length);
 
   if (ECMA_IS_VALUE_ERROR (new_typedarray))
   {
     return new_typedarray;
   }
 
-  ecma_object_t *new_obj_p = ecma_get_object_from_value (new_typedarray);
+  ecma_object_t *target_obj_p = ecma_get_object_from_value (new_typedarray);
+  ecma_typedarray_info_t target_info = ecma_typedarray_get_info (target_obj_p);
 
-  for (uint32_t index = 0; index < len && ecma_is_value_empty (ret_value); index++)
+  ecma_typedarray_getter_fn_t src_typedarray_getter_cb = ecma_get_typedarray_getter_fn (src_info.typedarray_id);
+  ecma_typedarray_setter_fn_t target_typedarray_setter_cb = ecma_get_typedarray_setter_fn (target_info.typedarray_id);
+
+  for (uint32_t index = 0; index < src_info.typedarray_length && ecma_is_value_empty (ret_value); index++)
   {
-    ecma_value_t current_index =  ecma_make_uint32_value (index);
-    ecma_value_t get_value = ecma_op_typedarray_get_index_prop (obj_p, index);
+    ecma_value_t current_index = ecma_make_uint32_value (index);
+
+    ecma_length_t src_byte_pos = (index << src_info.shift) + src_info.offset;
+    lit_utf8_byte_t *src_buff_p = ecma_arraybuffer_get_buffer (src_info.typedarray_buffer_p) + src_byte_pos;
+    ecma_number_t element_num = src_typedarray_getter_cb (src_buff_p);
+    ecma_value_t get_value = ecma_make_number_value (element_num);
     ecma_value_t call_args[] = { get_value, current_index, this_arg };
 
     ECMA_TRY_CATCH (mapped_value, ecma_op_function_call (func_object_p, cb_this_arg, call_args, 3), ret_value);
```

Comment:
Since this refactor is already affecting most of the typedarray builtins, we may as well remove the TRY_CATCH macros in one go.